notify-send.sh

```diff
@@ -200,109 +200,172 @@ process_special_action() {
     action_key="$1"
     command="$2"
 
-    if [[ -z "$action_key" ]] || [[ -z "$command" ]]; then
+    if [ -z "$action_key" ] || [ -z "$command" ]; then
         echo "Command must not be empty"
         exit 1
     fi
 
     ACTION_COMMANDS=("${ACTION_COMMANDS[@]}" "$action_key" "$command")
 
-    if [[ "$action_key" != close ]]; then
+    if [ "$action_key" != close ]; then
         local action="$(make_action "$action_key" "$name")"
         ACTIONS=("${ACTIONS[@]}" "$action")
     fi
 }
 
 process_posargs() {
-    if [[ "$1" = -* ]] && ! [[ "$positional" = yes ]] ; then
-        echo "Unknown option $1"
-        exit 1
+    case "$1" in
+        -*) if [ "$positional" != yes ] ; then
+                echo "Unknown option $1"
+                exit 1
+            fi ;;
+    esac
```

Comment:
Since we can't really do regular expression matching with the regular `test` or `[` `]`, we have to resort to other means to replace similar constructs.  Fortunately, `case` provides some fairly basic matching capabilities that we can benefit from.

We first check if `$1` is prefixed with `-` much in the same way as in the old `if` statement.  If that is the case, we check if `$positional` is not `yes`, using a familiar `!=` operator, meaning much the same as it does in other languages.

The difference between using `=` or `!=` and using `-e`, `-ne`, `-gt`, and co, is that the former apply only to strings while the latter are for numeric comparison.